src/main/groovy/edu/oregonstate/mist/trailsapi/resources/TrailResources.groovy

```diff
@@ -111,5 +113,28 @@ public class TrailsResource extends Resource {
         } else {
             notFound().build()
         }
+}
+
+    @PUT
+    @Path ('/{id: \\d+}')
+    @Consumes(MediaType.APPLICATION_JSON)
+    Response putTrail (@PathParam('id') Integer id, @Valid ResultObject newResultObject) {
+        Trail currentTrail = trailDAO.getTrailByID(id)
+        if (currentTrail) {
+            if (newResultObject) {
```

Comment:
Should you be running the `trailValidator` method on the updated trail before updating the DB?